### PR TITLE
{compiler}[GCCcore/10.3.0] LLVM v12.0.1

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by robert-mijakovic** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/4f3afb38ad0ef9faf8c3e76204b8d143 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.8.5
See https://gist.github.com/14a18c83ae4666330702f2ac1714bd3c for a full test report.

File: easybuild/easyconfigs/l/LLVM/LLVM-12.0.1-GCCcore-10.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+toolchainopts = {'cstd': 'gnu++11'}
+
+source_urls = ['https://github.com/llvm/llvm-project/releases/download/llvmorg-%(version)s/']
+sources = ['llvm-%(version)s.src.tar.xz']
+checksums = ['7d9a8405f557cefc5a21bf5672af73903b64749d9bc3a50322239f56f34ffddf']
+
+builddependencies = [
+    ('binutils', '2.36.1'),
+    ('CMake', '3.20.1'),
+    ('Python', '3.9.5'),
```
Comment (branfosj) [inline easybuild/easyconfigs/l/LLVM/LLVM-12.0.1-GCCcore-10.3.0.eb:26]:
```suggestion
    ('Python', '3.9.5', '-bare'),
```

We should use the less heavy `-bare` version of Python as the build dependency.

**Review by branfosj — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/l/LLVM/LLVM-12.0.1-GCCcore-10.3.0.eb (hunk 1 above)
Comment (robert-mijakovic) [reply easybuild/easyconfigs/l/LLVM/LLVM-12.0.1-GCCcore-10.3.0.eb]: Fixed!

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/d83ffe0202c555c018f221e41b51de4f for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0306u07a.bear.cluster - Linux RHEL 8.4, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/709f43d844097acc5619231408c9c8e6 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @robert-mijakovic!

**Pull request merged**